pkg/config/auth.go

```diff
@@ -0,0 +1,1162 @@
+		User       string `toml:"user"`
+		Pass       string `toml:"pass"`
+		AdminEmail string `toml:"admin_email"`
+		SenderName string `toml:"sender_name"`
+	}
+
+	emailTemplate struct {
+		Subject     string `toml:"subject"`
+		ContentPath string `toml:"content_path"`
+	}
+
+	sessions struct {
+		Timebox           time.Duration `toml:"timebox"`
+		InactivityTimeout time.Duration `toml:"inactivity_timeout"`
+	}
+
+	sms struct {
+		EnableSignup        bool              `toml:"enable_signup"`
+		EnableConfirmations bool              `toml:"enable_confirmations"`
+		Template            string            `toml:"template"`
+		Twilio              twilioConfig      `toml:"twilio" mapstructure:"twilio"`
+		TwilioVerify        twilioConfig      `toml:"twilio_verify" mapstructure:"twilio_verify"`
+		Messagebird         messagebirdConfig `toml:"messagebird" mapstructure:"messagebird"`
+		Textlocal           textlocalConfig   `toml:"textlocal" mapstructure:"textlocal"`
+		Vonage              vonageConfig      `toml:"vonage" mapstructure:"vonage"`
+		TestOTP             map[string]string `toml:"test_otp"`
+		MaxFrequency        time.Duration     `toml:"max_frequency"`
+	}
+
+	twilioConfig struct {
+		Enabled           bool   `toml:"enabled"`
+		AccountSid        string `toml:"account_sid"`
+		MessageServiceSid string `toml:"message_service_sid"`
+		AuthToken         string `toml:"auth_token" mapstructure:"auth_token"`
+	}
+
+	messagebirdConfig struct {
+		Enabled    bool   `toml:"enabled"`
+		Originator string `toml:"originator"`
+		AccessKey  string `toml:"access_key" mapstructure:"access_key"`
+	}
+
+	textlocalConfig struct {
+		Enabled bool   `toml:"enabled"`
+		Sender  string `toml:"sender"`
+		ApiKey  string `toml:"api_key" mapstructure:"api_key"`
+	}
+
+	vonageConfig struct {
+		Enabled   bool   `toml:"enabled"`
+		From      string `toml:"from"`
+		ApiKey    string `toml:"api_key" mapstructure:"api_key"`
+		ApiSecret string `toml:"api_secret" mapstructure:"api_secret"`
+	}
+
+	thirdParty struct {
+		Firebase tpaFirebase `toml:"firebase"`
+		Auth0    tpaAuth0    `toml:"auth0"`
+		Cognito  tpaCognito  `toml:"aws_cognito"`
+	}
+
+	tpaFirebase struct {
+		Enabled bool `toml:"enabled"`
+
+		ProjectID string `toml:"project_id"`
+	}
+
+	tpaAuth0 struct {
+		Enabled bool `toml:"enabled"`
+
+		Tenant       string `toml:"tenant"`
+		TenantRegion string `toml:"tenant_region"`
+	}
+
+	tpaCognito struct {
+		Enabled bool `toml:"enabled"`
+
+		UserPoolID     string `toml:"user_pool_id"`
+		UserPoolRegion string `toml:"user_pool_region"`
+	}
+)
+
+func sha256Hmac(key, value string) string {
+	h := hmac.New(sha256.New, []byte(key))
+	h.Write([]byte(value))
+	return hex.EncodeToString(h.Sum(nil))
+}
+
+func compareHashedField(secret string, local, remote *string) {
+	if remote == nil {
+		return
+	}
+	if *local == "" && *remote == "" {
+		return
+	}
+	localHashed := sha256Hmac(secret, *local)
+	diff.CompareSensitiveField(&localHashed, remote)
```

Comment:
**remark**

Here I still use `CompareSentitiveField` to keep a human readable result message. We could just use the standard comparaison over the hashed values but I think that might confuse the users wondering why the value they set in their config get a different value in their config diff.